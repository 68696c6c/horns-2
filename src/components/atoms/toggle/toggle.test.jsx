/* global describe, it, expect */
import React from 'react'

import { render } from 'setup-test'

import Toggle from '.'

describe('Toggle', () => {
  it('should render as a checkbox', () => {
    const { container } = render(
      <Toggle type="checkbox" id="example" name="example" />
    )
    expect(container).toMatchSnapshot()
  })
  it('should render as a radio', () => {
    const { container } = render(
      <Toggle type="radio" id="example" name="example" />
    )
    expect(container).toMatchSnapshot()
  })
})
