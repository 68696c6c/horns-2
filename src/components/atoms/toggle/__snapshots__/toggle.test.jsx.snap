// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render as a checkbox 1`] = `
.emotion-0 {
  display: none;
}

.emotion-0:checked + label.control {
  background: rgb(216,216,216);
}

.emotion-0:disabled + label.control {
  background: rgb(191,191,191);
  cursor: not-allowed;
}

.emotion-0:disabled + label.control + label.toggle-label {
  cursor: not-allowed;
}

.emotion-1 {
  border-top-width: 2px;
  border-top-style: solid;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-left-width: 2px;
  border-left-style: solid;
  border-right-width: 2px;
  border-right-style: solid;
  background: rgb(255,255,255);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
  cursor: pointer;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: content-box;
  vertical-align: middle;
  content: ' ';
}

.emotion-1:disabled,
.emotion-1.inactive {
  cursor: pointer;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):hover {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):active {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

<div>
  <input
    class="toggle emotion-0"
    cursor="pointer"
    id="example"
    name="example"
    placeholder=""
    type="checkbox"
  />
  <label
    class="control emotion-1"
    cursor="pointer"
    for="example"
    name="example"
    placeholder=""
    type="checkbox"
  />
</div>
`;

exports[`Toggle should render as a radio 1`] = `
.emotion-0 {
  display: none;
}

.emotion-0:checked + label.control {
  background: rgb(216,216,216);
}

.emotion-0:disabled + label.control {
  background: rgb(191,191,191);
  cursor: not-allowed;
}

.emotion-0:disabled + label.control + label.toggle-label {
  cursor: not-allowed;
}

.emotion-1 {
  border-top-width: 2px;
  border-top-style: solid;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-left-width: 2px;
  border-left-style: solid;
  border-right-width: 2px;
  border-right-style: solid;
  background: rgb(255,255,255);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
  cursor: pointer;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: content-box;
  vertical-align: middle;
  content: ' ';
  border-radius: 50%;
}

.emotion-1:disabled,
.emotion-1.inactive {
  cursor: pointer;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):hover {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):active {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

<div>
  <input
    class="toggle emotion-0"
    cursor="pointer"
    id="example"
    name="example"
    placeholder=""
    type="radio"
  />
  <label
    class="control emotion-1"
    cursor="pointer"
    for="example"
    name="example"
    placeholder=""
    type="radio"
  />
</div>
`;

exports[`Toggle should render as default 1`] = `
.emotion-0 {
  display: none;
}

.emotion-0:checked + label.control {
  background: rgb(216,216,216);
}

.emotion-0:disabled + label.control {
  background: rgb(191,191,191);
  cursor: not-allowed;
}

.emotion-0:disabled + label.control + label.toggle-label {
  cursor: not-allowed;
}

.emotion-1 {
  border-top-width: 2px;
  border-top-style: solid;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-left-width: 2px;
  border-left-style: solid;
  border-right-width: 2px;
  border-right-style: solid;
  background: rgb(255,255,255);
  color: rgb(1,1,1);
  border-color: rgb(216,216,216);
  cursor: pointer;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  font-family: Helvetica,sans-serif;
  font-size: medium;
  font-style: normal;
  font-weight: 400;
  line-height: 1em;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  box-sizing: content-box;
  vertical-align: middle;
  content: ' ';
}

.emotion-1:disabled,
.emotion-1.inactive {
  cursor: pointer;
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):hover {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

.emotion-1:not(:disabled):not(.inactive):active {
  -webkit-text-decoration-line: none;
  text-decoration-line: none;
}

<div>
  <input
    class="toggle emotion-0"
    cursor="pointer"
    id="example"
    name="example"
    placeholder=""
    type="checkbox"
  />
  <label
    class="control emotion-1"
    cursor="pointer"
    for="example"
    name="example"
    placeholder=""
    type="checkbox"
  />
</div>
`;
